python/lsst/drp/tasks/assemble_cell_coadd.py

```diff
@@ -113,14 +113,15 @@ class AssembleCellCoaddConfig(PipelineTaskConfig, pipelineConnections=AssembleCe
         inclusiveMin=True,
         inclusiveMax=False,
     )
-    # The following config options are specific to the CoaddPsf.
-    coadd_psf = ConfigField(
-        doc="Configuration for CoaddPsf",
-        dtype=CoaddPsfConfig,
+    psf_warper = ConfigField(
+        doc="Configuration for the warper that warps the PSFs. It must have the same configuration used to "
+        "warp the images.",
```

Comment:
We will have to enforce this via a contract in the pipeline, since I don't think there's any other way to enforce consistency across tasks.

Reply:
I agree that's the best approach for now, but I think it's also a nudge in the direction of moving PSF warping down to `MakeDirectWarpTask` (by having that warp yet another image plane that would for now have to be yet another separate dataset type).  I don't think we should consider doing that on this ticket, but between configuration consistency, free parallelism over visits, and ruling out differences in how the WCS is interpolated during warping, I think that's the direction we want to go eventually.

Reply:
I see that to be the natural progression, given that we've moved PSF evaluation from measurement to coaddition and the next natural step is to move it to `make_direct_warp.py`. We spoke about this, but my thought here has been to use `MultipleCellCoadd` (we might need to subclass it, more on that later) with no buffer between cells to be the data structure to hold warps. It has the capability to hold multiple noise image planes which are used by cell-based coadds as well. The PSF images can be evaluated at the cell centers at the time of warping.

The reason why we might need another class - or make a generic parent class to subclass from - is that we now have to allow for multiple detectors within each cell (we don't want to discard them just yet). This needs some minor changes to how we serialize metadata info.

But to reiterate again,
> differences in how the WCS is interpolated during warping

is not an issue because we evaluate the PSF image on the calexp and warp them here.

Reply:
> But to reiterate again,
>>    differences in how the WCS is interpolated during warping
>
> is not an issue because we evaluate the PSF image on the calexp and warp them here.

We don't evaluate the PSF image on the _full_ calexp; we evaluate it into a much smaller image in calexp coordinates.  While that would be sufficient if warping used the real WCS for every pixel, or if it evaluated the WCS at fixed points in PARENT coordinates (accounting for xy0) and interpolated between them.  But I suspect that it might actually evaluate the WCS at fixed points in LOCAL coordinates (i.e. not accounting for xy0) and interpolate those.  Just suspicion, though; if you know that it evaluates the PSF at fixed points in PARENT coordinates then this concern is indeed unfounded.

Reply:
Isn't this related to your comment about `xy0` below? The bounding box of the PSF image is evaluated in PARENT coordinates by default but I wonder if I'm missing your point. Happy to discuss this more tomorrow.

Reply:
This question doesn't need to hold up this branch anyway, and I think the answer is one that can be found in the depths of the warping code: we're making different assumptions about what it's doing, but the truth is out there.